minerl/env/malmo.py

```diff
@@ -508,53 +509,52 @@ def launch(self):
                     raise RuntimeError("Malmo failed to start the MalmoEnv server! Check the logs from the Minecraft process.");
                 self._logger.info("Minecraft process ready")
                  
-
                 if not port == self._port:
                     self._logger.warning("Tried to launch Minecraft on port {} but that port was taken, instead Minecraft is using port {}.".format(port, self.port))
+
                 # supress entire output, otherwise the subprocess will block
                 # NB! there will be still logs under Malmo/Minecraft/run/logs
                 # FNULL = open(os.devnull, 'w')
                 # launch a logger process
-                def log_to_file(logdir):
-                    if not os.path.exists(os.path.join(logdir, 'logs')):
-                            os.makedirs((os.path.join(logdir, 'logs')))
+                def malmo_java_stdout_log_to_file(logdir):
+                    os.makedirs(os.path.join(logdir, 'logs'), exist_ok=True)
+                    log_path = os.path.join(logdir, 'logs',
+                                            'mc_{}.log'.format(self._target_port))
+                    self._logger.info("Logging Minecraft+Malmo java instance to {}".format(log_path))
+
+                    MB = 1024**2
+                    direct_format = logging.Formatter('%(message)s')
+                    log_handler = logging.handlers.RotatingFileHandler(
+                        log_path, mode='w', maxBytes=5 * MB,
```

Comment:
Won't this lead to max filesize being 5 MB? Can we make MAX_MB a variable for legibility's sake?

Reply:
Every time the log file grows to 5 MB a log rotation occurs where the current log file is moved to `${log_file_name}.1`, `${log_file_name}.1` is moved to ...2 if it exists, and ...2 is deleted. This is what the backupCount parameter means.

Reply:
> Can we make MAX_MB a variable for legibility's sake?
Sure, I'll add it to the function as a default parameter.

Reply:
Doing the rotation is more complicated than just deleting the file and starting over, but it has the advantage that won't crash right after deleting the file and end up having no logs.